src/app/clients/getClientsData.ts

```diff
@@ -33,8 +34,14 @@ const getClientsData = async (
 
     query = query.range(startIndex, endIndex);
 
+    query = query.abortSignal(abortSignal);
+
     const { data: clients, error: clientError } = await query;
 
+    if (abortSignal.aborted) {
+        return { data: [], abortSignalResponse: abortSignal };
+    }
```

Comment:
yes it does, but after this line, we check for errors in the response and throw an error if there is one. This is bad because if we abort, we don't want to throw an error. so we need to return from the fetch function before. 

Another option that I tried was not throwing an error if there is a specific "abort" error, but the abort error returned by supabase was inconsistent. 

Plus, we have to return something from the fetch function, so if we were to return null or [] or something, we'd have to have code in ParcelsPage.tsx checking if the data returned from the fetch function was truthy, and if so, set the data to be that. I thought that was less readable than returning the abortSignal, which allows us to check if the abortSignal.aborted is true

thoughts?

Reply:
Interesting - I would've expected the error returned by supabase to have a consistent code or something at least.

I understand your comment about dealing with the error just after this. IMO the real problem lies in the lack of error handling by this function's caller. If this fetching fails somehow, we should handle it gracefully on the frontend (maybe _not doing anything_ is what we want in the `catch`). 

It'd then be fine to throw an error here even when the request is aborted. (obviously it'd be much better if we could log something else when aborted, but if we can't meaningfully distinguish between an aborted request and other errors, that's fine)

Reply:
Okay cool I agree we should be dealing with it gracefully in ParcelsPage.tsx makes sense. 

Are you sure its ok to throw an error when the request is aborted? because we're expecting that to happen quite a lot of the time, so it would really clog up the cloudwatch log.

Reply:
Ok sure good idea. Have done